src/crs_linter/rules/rule_tests.py

```diff
@@ -0,0 +1,48 @@
+from crs_linter.lint_problem import LintProblem
+from crs_linter.rule import Rule
+
+
+class RuleTests(Rule):
+    """Check that rules have corresponding test cases."""
+
+    def __init__(self):
+        super().__init__()
+        self.success_message = "All rules have tests."
+        self.error_message = "There are one or more rules without tests."
+        self.error_title = "no tests"
+        self.args = ("data", "test_cases", "exclusion_list")
+
+    def check(self, data, test_cases=None, exclusion_list=None):
+        """
+        Check that rules have corresponding test cases
+        """
+        if test_cases is None:
+            test_cases = {}
+        if exclusion_list is None:
+            exclusion_list = []
+        
+        for d in data:
+            # only SecRule counts
+            if d['type'] == "SecRule":
+                for a in d['actions']:
```

Comment:
Here you don't need to iterate through all actions, just use your implemented `get_id()`. I think that would make the code a bit more clear. (But you will need the id in two formats: int and string)